crates/noirc_evaluator/src/ssa/conditional.rs

```diff
@@ -491,50 +491,50 @@ impl DecisionTree {
     ) -> Vec<NodeId> {
         // 1. find potential matches between the two blocks
         let mut candidates = Vec::new();
-        let l_iter = left.iter().enumerate().filter(|&i| {
-            let ins = ctx.get_instruction(*i.1);
+        let keep_call_and_store = |node_id: NodeId| -> bool {
+            let ins = ctx.get_instruction(node_id);
             matches!(ins.operation.opcode(), Opcode::Call(_) | Opcode::Store(_))
-        });
-        let mut r_iter = right.iter().enumerate().filter(|&i| {
-            let ins = ctx.get_instruction(*i.1);
-            matches!(ins.operation.opcode(), Opcode::Call(_) | Opcode::Store(_))
-        });
-        for i in l_iter {
-            let ins = ctx.get_instruction(*i.1);
-            for j in r_iter.by_ref() {
-                //while let Some(j) = r_iter.next() {
-                let jins = ctx.get_instruction(*j.1);
-                match &jins.operation {
-                    Operation::Call { func_id, returned_arrays, .. } => {
-                        // we do not support returned arrays for now
-                        if returned_arrays.is_empty() {
-                            if let Operation::Call { func_id: f2, returned_arrays: r2, .. } =
-                                &ins.operation
-                            {
-                                if r2.is_empty() && f2 == func_id {
-                                    candidates.push(Segment::new(i, j));
-                                }
-                            }
+        };
+        let l_iter = left.iter().enumerate().filter(|&i| keep_call_and_store(*i.1));
+        let mut r_iter = right.iter().enumerate().filter(|&i| keep_call_and_store(*i.1));
+        for left_node in l_iter {
+            let left_ins = ctx.get_instruction(*left_node.1);
+            for right_node in r_iter.by_ref() {
+                let right_ins = ctx.get_instruction(*right_node.1);
+                match (&left_ins.operation, &right_ins.operation) {
+                    (
+                        Operation::Call {
+                            func_id: left_func, returned_arrays: left_arrays, ..
+                        },
+                        Operation::Call {
+                            func_id: right_func, returned_arrays: right_arrays, ..
+                        },
+                    ) => {
+                        if left_func == right_func
```

Comment:
These ifs can be a pattern guard as well `( Operation::Call .... ) if left_func ... => condidates...`.
Its very minor, saves 1 level of nesting. Thought I'd mention it to explore it but maybe cargo fmt will make it look worse in the end.

Reply:
Indeed cargo fmt is not doing a great job here, it is even inconsistent, but yet I prefer the output with the pattern guard.